File: src/InvestmentPerformanceService/InvestmentPerformance.Domain/AggregatesModel/InvestmentAggregate/InvestmentDetail.cs
using InvestmentPerformance.Domain.AggregatesModel.InvestmentAggregate;
using InvestmentPerformance.Domain.SeedWork;

namespace InvestmentPerformance.Domain.AggregatesModel
{
    public class InvestmentDetail : Entity
    {
        protected InvestmentDetail() {}

        public int InvestmentId { get; private set; }

        public decimal CostBasisPerShare { get; private set; }

        public int CurrentValue { get; private set; }

        public decimal CurrentPrice { get; private set; }

        public Term Term { get; private set; }

        public decimal TotalGainOrLoss { get; private set; }
    }
}

File: InvestmentTradingPlatform/src/main/java/com/mm4nn/InvestmentTradingPlatform/domain/bean/InvestmentDetail.java
package com.mm4nn.InvestmentTradingPlatform.domain.bean;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 *  POJO used for encapsulation of investment data
 */
@Setter
@Getter
@EqualsAndHashCode
@ToString
@NoArgsConstructor
public class InvestmentDetail {

    private Stock stock;

    private Integer numberOfShares;
    private BigDecimal pricePerShare;

    private LocalDateTime purchaseDate;

    private BigDecimal costBasisPerShare;
    private BigDecimal currentPrice;
    private BigDecimal initialValue;
    private BigDecimal currentValue;
    private InvestmentTerm term;
    private BigDecimal totalLossOrGain;

    public InvestmentDetailRepresentation toInvestmentDetailRepresentation() {
        return new InvestmentDetailRepresentation(numberOfShares, costBasisPerShare,
                currentPrice, currentValue, term, totalLossOrGain);
    }

}
